depsync/depsync.go

```diff
@@ -159,18 +233,26 @@ FETCH:
 		} else if err != nil {
 			return err
 		}
-		fmt.Printf("more=%v, cursor=%s, fetched=%v\n", resp.MoreToFollow, resp.Cursor, resp.FetchedUntil)
+		level.Info(w.logger).Log("msg", "DEP fetch", "more", resp.MoreToFollow, "cursor", resp.Cursor, "fetched", resp.FetchedUntil, "devices", len(resp.Devices))
 		w.conf.Cursor = cursor{Value: resp.Cursor, CreatedAt: time.Now()}
 		if err := w.conf.Save(); err != nil {
 			return errors.Wrap(err, "saving cursor from fetch")
 		}
-		e := NewEvent(resp.Devices)
-		data, err := MarshalEvent(e)
-		if err != nil {
-			return err
-		}
-		if err := w.publisher.Publish(context.TODO(), SyncTopic, data); err != nil {
-			return err
+		if len(resp.Devices) > 0 {
+			e := NewEvent(resp.Devices)
+			data, err := MarshalEvent(e)
+			if err != nil {
+				return err
```

Comment:
errors.Wrap(err, .. somecontext) would be great here.

Reply:
👍